### ISSUE TITLE
Enforce "dataSource" spelling via a Checkstyle regex

### ISSUE THREAD

**Issue by leventov** [apache/incubator-druid#6870]
Currently the codebase is a free mix of "datasource" and "dataSource" spelling.

**Comment by gianm** [comment]
The unwritten style rule I have been following is that in JSON and in code (method/variable names) the proper spelling is "dataSource" or "DataSource". But in natural language documentation, the proper spelling is "datasource", because writing dataSource or DataSource looks strange in the middle of natural language text.

If we were starting from scratch, I'd say call them "datasources" everywhere, but it'd be too disruptive to change all the JSON and code.

**Comment by leventov** [comment]
Good point. I thought that proper spelling in comments is "data source" (with a space), but now I'm not sure.

Maybe the scope of regex could be reduced to Java identifiers. (If not via Checkstyle, I'm pretty sure it's possible with IntelliJ structural search).
